ui/utils.py

```diff
@@ -16,3 +16,22 @@ def wrapper(self, *args, **kwargs):
 
 def clamp(value, _min, _max):
     return max(_min, min(value, _max))
+
+
+def check_value_lock(func):
+    # A safety check wrapper so that there's no race conditions between functions able to change position/value
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        assert self.value_lock, "Class has no member self.value_lock"  # todo:maybe we should create it here ?
```

Comment:
should we create the lock if it doesn't exist ?

Reply:
I think we shouldn't. Much like the `to_be_foreground` relies on `to_foreground` to be present, I think this function has to rely on `value_lock` to be present. However, I'm not sure - I think I'll ask a StackOverflow question, and let's make it not create the lock in the meanwhile =)

Reply:
Sure, add the link to the discussion when it's done I'm interested !
(or maybe codereview.stackexchange is more relevant ?)

Reply:
Yep, codereview would definitely be more relevant =) For example, [I asked this recently](https://codereview.stackexchange.com/questions/179711/generic-oneshot-task-runner) when I was unsure about other class that I wrote.